### ISSUE TITLE
Only the first type selector in a selector list is ignored when ignore: ["descendant"] is used in selector-no-type

### ISSUE THREAD

**Issue by lichenbuliren** [stylelint/stylelint#1808]
> Describe the issue. Is it a bug or a feature request (new rule, new option, etc.)?

e.g. "A bug when multiple el selector with comma separated"

> Which rule, if any, is this issue related to?

e.g. `selector-no-type`

> What CSS is needed to reproduce this issue?

e.g.

```css
.info-row{
  overflow: hidden;

  dt,dd {
    float: left;
  }

  dt {
    opacity: .5;
    margin-right: px2rem(40);
  }
}
```

> What stylelint configuration is needed to reproduce this issue?

e.g. default config

> Which version of stylelint are you using?

e.g. `7.0.3`

> How are you running stylelint: CLI, PostCSS plugin, Node API?

e.g. "CLI with `stylelint --config myconfig *.css`"

> Does your issue relate to non-standard syntax (e.g. SCSS, nesting, etc.)?

e.g. "Yes, it's related to SCSS maps."

> What did you expect to happen?

e.g. "No warnings to be flagged."

> What actually happened (e.g. what warnings or errors you are getting)?

e.g. `Unexpected type selector   selector-no-type`

**Comment by jeddy3** [comment]
@lichenbuliren Thanks for writing up the report.

The `selector-no-type` rules seems to be behaving as expected i.e your code snippet gives the following warnings:

```console
4:3 error Unexpected type selector (selector-no-type)
4:6 error Unexpected type selector (selector-no-type)
8:3 error Unexpected type selector (selector-no-type)
```

There are three instances of *type* selectors (2x `dt` and `dd`) in your code snippet and each is correctly flagged at the correct line and column.

Can you help us by explaining a little more about why you weren't expecting this behaviour?

**Comment by lichenbuliren** [comment]
if i change code like this: 
``` css
```

**Comment by lichenbuliren** [comment]
if i change my code like this, it's works well:
```css
.info-row{
    overflow: hidden;

    dd {
      float: left;
    }

    dt {
      float: left;
      opacity: .5;
      margin-right: px2rem(40);
    }
  }
```
my .stylelintrc.yaml is : 

``` yaml
rules:
  color-no-invalid-hex: true
  font-family-name-quotes: always-where-recommended
  function-blacklist:
  - calc
  function-name-case: lower
  property-no-vendor-prefix: true
  declaration-no-important: true
  declaration-block-no-duplicate-properties: true
  declaration-block-semicolon-newline-after: always
  declaration-block-trailing-semicolon: always
  block-closing-brace-newline-after: always
  block-opening-brace-newline-after: always
  block-no-empty: true
  selector-max-compound-selectors: 4
  selector-no-qualifying-type: true
  selector-no-type:
  - true
  - ignore:
    - descendant
```

**Comment by jeddy3** [comment]
@lichenbuliren Thanks. I understand now. You're using the `"ignore": ["descendant"]` optional secondary option.

Yes, there appears to be a bug.

Given:

```console
"selector-no-type": [true, { "ignore": ["descendant"] }],
```

And:

```css
.info-row{
  overflow: hidden;

  dt,dd {
    float: left;
  }

  dt {
    opacity: .5;
    margin-right: px2rem(40);
  }
}
```

The following warning is erroneously flagged:

```console
5:6 error Unexpected type selector (selector-no-type)
```

I'm going to flag this as a bug and label is as "help wanted". Would you like to [contribute](https://github.com/stylelint/stylelint/blob/master/docs/developer-guide/rules.md#fixing-bugs) a fix that ensures that all the *type selectors* in a *selector list* are ignored when the `"ignore": ["descendant"]` optional secondary option used? At the moment it appears that only the first type selector in a list is ignored.
